airbyte-workers/src/main/java/io/airbyte/workers/internal/EmptyAirbyteSource.java

```diff
@@ -4,30 +4,90 @@
 
 package io.airbyte.workers.internal;
 
+import com.fasterxml.jackson.databind.JsonNode;
 import io.airbyte.commons.json.Jsons;
+import io.airbyte.config.ResetSourceConfiguration;
+import io.airbyte.config.StreamDescriptor;
 import io.airbyte.config.WorkerSourceConfig;
+import io.airbyte.protocol.models.AirbyteGlobalState;
 import io.airbyte.protocol.models.AirbyteMessage;
 import io.airbyte.protocol.models.AirbyteMessage.Type;
 import io.airbyte.protocol.models.AirbyteStateMessage;
+import io.airbyte.protocol.models.AirbyteStateMessage.AirbyteStateType;
+import io.airbyte.protocol.models.AirbyteStreamState;
+import io.airbyte.protocol.models.StateMessageHelper;
+import io.vavr.control.Either;
 import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.EmptyStackException;
+import java.util.LinkedList;
+import java.util.List;
 import java.util.Optional;
+import java.util.Queue;
+import java.util.Stack;
 import java.util.concurrent.atomic.AtomicBoolean;
+import lombok.extern.slf4j.Slf4j;
 
 /**
  * This source will never emit any messages. It can be used in cases where that is helpful (hint:
  * reset connection jobs).
  */
+@Slf4j
 public class EmptyAirbyteSource implements AirbyteSource {
 
   private final AtomicBoolean hasEmittedState;
+  private final Queue<StreamDescriptor> streamDescriptors = new LinkedList<>();
+  private boolean isPartialReset;
+  private boolean isStarted = false;
+  private Either<List<AirbyteStateMessage>, AirbyteStateMessage> perStreamOrGlobalState;
 
   public EmptyAirbyteSource() {
     hasEmittedState = new AtomicBoolean();
   }
 
   @Override
   public void start(final WorkerSourceConfig sourceConfig, final Path jobRoot) throws Exception {
-    // no op.
+
+    try {
+      if (sourceConfig == null || sourceConfig.getSourceConnectionConfiguration() == null) {
+        isPartialReset = false;
+      } else {
+        ResetSourceConfiguration sourceConfiguration = Jsons.object(sourceConfig.getSourceConnectionConfiguration(), ResetSourceConfiguration.class);
+        streamDescriptors.addAll(sourceConfiguration.getStreamsToReset());
+        if (streamDescriptors.isEmpty()) {
+          isPartialReset = false;
```

Comment:
I believe this is to allow this reset source to work with our existing logic of full connection resets, which is persisting a `ResetSourceConfiguration` with empty `streamsToReset` since we currently don't have the logic to add streams to the stream_reset table. 

However, once we have fully migrated to persisting all streams to the stream_reset table in the full connection reset case, we will probably want this to throw an exception in this case instead, since we wouldn't expect this to be called with an empty set of stream descriptors at that point. 

Could you add a TODO comment here indicating that we want to change this, and link a brief issue explaining this change?